extensions/cache/runtime-spi/src/main/java/io/quarkus/cache/Cache.java

```diff
@@ -42,6 +43,8 @@ public interface Cache {
      */
     <K, V> Uni<V> get(K key, Function<K, V> valueLoader);
 
+    <K, V> Uni<V> get(K key, Function<K, V> valueLoader, Duration expiresIn);
```

Comment:
We'll need a proper javadoc here.

Reply:
what signature do you imagine with this expiry object ? 
something like 

`<K, V> Uni<V> get(K key, Function<K, V> valueLoader, Expiry expiry);` ?

I think it has the benefit of allowing a more fine grained control (expire after create, expire after update)

However the expiration must be computed every time for a read, we could avoid that by returning it from the value loader but I am not sure how to do it since it would have the same erasure. maybe with a new method name ?